test/e2e/density.go

```diff
@@ -252,13 +259,13 @@ func cleanupDensityTest(dtc DensityTestConfig) {
 		namespace := dtc.Configs[i].GetNamespace()
 		kind := dtc.Configs[i].GetKind()
 		// TODO: Remove Deployment guard once GC is implemented for Deployments.
-		if framework.TestContext.GarbageCollectorEnabled && kind != extensions.Kind("Deployment") {
+		if framework.TestContext.GarbageCollectorEnabled && kind != extensions.Kind("Deployment") && kind != batch.Kind("Job") {
```

Comment:
Couldn't this condition be exported to a separate function with a meaningful name ?